java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -346,6 +371,39 @@ private void build(Tree tree) {
     }
   }
 
+  private void buildBlockStatement(BlockTree tree) {
+    Block finallyBlock = finallyBlocks.peek();
+    if (catchPoints.isEmpty() && finallyBlock == null) {
+      for (StatementTree statementTree : Lists.reverse((tree).body())) {
+        build(statementTree);
+      }
+    } else {
+      buildBlockStatementWithinTryBlock(tree, finallyBlock);
+    }
+  }
+
+  private void buildBlockStatementWithinTryBlock(BlockTree tree, Block finallyBlock) {
+    for (StatementTree statementTree : Lists.reverse((tree).body())) {
+      currentBlock = createBlock(currentBlock);
+      build(statementTree);
+      for (CatchPoint catchBlock : catchPoints) {
+        if (canRaise(currentBlock, catchBlock.exception())) {
+          currentBlock.successors().add(catchBlock.block());
+        }
+      }
+      if (finallyBlock != null) {
+        currentBlock.successors().add(finallyBlock);
+      }
+    }
+  }
+
+  private boolean canRaise(Block block, SyntaxToken exception) {
+    // FIXME this method should return true if the block can raise the supplied exception
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1134)